### ISSUE TITLE
[BUG] Existing World attempting to modify world.deployable compendium

### ISSUE THREAD

**Issue by msprijatelj** [Eranziel/foundryvtt-lancer#517]
**Describe the bug**
On startup in an existing Lancer world with actors that use Deployables present in the Actors tab, Foundry throws errors claiming `You may not modify the world.deployable Compendium which is currently locked.`

**To Reproduce**
1. Have a Lancer world on Lancer v1.4.0-beta.1, Foundry v10
2. Create a new pilot, and import a pilot from CompCon that has a mech with deployables
3. Log out of the world. 
4. Update to Lancer v1.4.2.
5. Open the same world from Step 1.
6. Observe that several error banners are thrown regarding modification of world.deployable.

**Expected behavior**
No errors are thrown or attempts made to modify the world.deployable Compendium on startup.

**Screenshots**
![image](https://user-images.githubusercontent.com/44978587/214098702-d92ddd02-1f67-4be7-bc99-eee3fcdf0030.png)

**Platform (please complete the following information):**
 - OS: Windows 10
 - Browser: Foundry App
 - Foundry version 10.291
 - Lancer system version 1.4.2

**Additional context**
This does not happen on a new world, even after importing pilots that have deployables. 

Not sure if this is due to data migration issues from Lancer ~~v1.3.1 or~~ v1.4.0-beta.1, or due to bad imports I attempted during Lancer v1.4.1.

Deleting all Deployable-type actors in the world didn't fix the issue, but deleting some duplicate mechs (including around 7-ish "New Mechs" which may have been created in preparation for v1.4.1 imports? not sure where they came from) led the issue to resolve itself.  I might suggest simply cleaning up some mech actor imports to resolve this issue, but I'd appreciate verification.

UPDATE: Just migrated an old v9 Lancer world to v10 Lancer v1.4.2, no errors regarding `world.deployables`. I'm inclined to believe it's leftovers from the beta/v1.4.1.

**Discord contact info**
Valkyrion#6196

**Comment by msprijatelj** [comment]
UPDATE: I can sortof reproduce this in an existing Lancer world that has an older version of Lancer Compendium Data:
1. Create a pilot in CompCon that has deployables
2. Create a share code for that CompCon Pilot
3. Create a new pilot in Foundry, import the CompCon pilot into this Foundry pilot via share code
4. Observe a bunch of errors pop up for "you may not modify the world.deployable Compendium" as mentioned above:
![image](https://user-images.githubusercontent.com/44978587/215543119-b3f85acf-0fba-4348-a001-7c7f613e4020.png)

This only happens on pilot import, but **_not_** on page refresh like the original issue. The installed LANCER Compendium is v3.0.42.

At @dodgepong 's suggestion, I tried updating the LANCER compendium to the latest version (v3.0.46). Attempting to import via share code no longer throws the `world.deployable` errors, so I would recommend updating the LANCER compendium as a workaround.

**Comment by msprijatelj** [comment]
Okay, on another world I updated from v9 to v10: I hit `world.deployable` errors when importing a pilot using core data, but after I installed the Long Rim LCP (completely fresh, did not exist previously), future imports on the old pilot using core data no longer threw errors. There may be some sort of "cleanup" occurring on both Core Data rebuilding and LCP installation? Which is good, to be fair, it just threw me for a second.
